client/wrappers/retryable/wrappers_test.go

```diff
@@ -0,0 +1,73 @@
+// The MIT License (MIT)
+
+// Copyright (c) 2017-2020 Uber Technologies Inc.
+
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in all
+// copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+
+package retryable
+
+import (
+	"context"
+	"testing"
+
+	"github.com/golang/mock/gomock"
+	"github.com/stretchr/testify/assert"
+
+	"github.com/uber/cadence/client/frontend"
+	"github.com/uber/cadence/common"
+	"github.com/uber/cadence/common/types"
+)
+
```

Comment:
would be nice to cover the new generated clients here as you did in common/persistence/wrappers/ratelimited/wrappers_test.go

Reply:
My idea is that there is no special logic in methods, so there is no point of testing every single method. Ive tested one, the rest are the same.
Persistent clients are harder, since it is more complicated and has extra logs/metrics. Though, tests there also could be cleaned up.

Reply:
Yeah agree there's not too much value in testing all methods. I wonder if we should put `gen/` somewhere in the path for generated files so we an easily exclude these from coverage reports for good.

Reply:
Usually, all go tooling respects generated headers (which gowrap writes)
https://github.com/golang/lint/issues/239
Coveralls respects it: https://coveralls.io/github/uber/cadence (.gen folder is not counted in lines to cover).
My assumption is that sonarqube also does, but might be we need to manually set it up.

Reply:
We manually set an exceptions:
https://github.com/uber/cadence/blob/master/Makefile#L601
So we can define a simple rule to separate generated files from the rest.

Reply:
+1 for `_generated.go` suffix unless the files are under .gen folder already. Ping me if you decide to go with that and I will update the internal coverage exclusions accordingly

Reply:
fwiw I'm kinda fond of this sort of structure:
```
dir/
  - thing_gen.go  // only trivially-correct codegen
  - thing.go      // anything that "completes" ^_gen, e.g. helper funcs
  - thing_test.go // tests of thing.go
```
_gen vs _generated: no preference.  I see _gen more but both are clear and that's all I really care about.

for code coverage and IDE highlighting purposes, many things look for a `// @generated` comment, which these don't seem to have and might be worth adding in general.  I'm not sure if we have anything that expects that or if it'd be easier to just filter them out in the makefile by filename suffix tho.

Reply:
hmm I'm not sure about @generated comment. 
I think, IDE/Linters looks for standard Go comment 
```
^// Code generated .* DO NOT EDIT\.$
```
https://pkg.go.dev/cmd/go#hdr-Generate_Go_files_by_processing_source